onnxruntime/core/providers/dnnl/dnnl_provider_factory.cc

```diff
@@ -1,38 +1,52 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+#include "core/providers/shared_library/provider_author.h"
 #include "core/providers/dnnl/dnnl_provider_factory.h"
 #include <atomic>
 #include "dnnl_execution_provider.h"
-#include "core/session/abi_session_options_impl.h"
 
 using namespace onnxruntime;
 
+extern onnxruntime::ProviderHost* g_host;
+
 namespace onnxruntime {
-struct DnnlProviderFactory : IExecutionProviderFactory {
+
+void SetProviderHost(ProviderHost& host);
+
+struct DnnlProviderFactory : Provider_IExecutionProviderFactory {
   DnnlProviderFactory(bool create_arena) : create_arena_(create_arena) {}
   ~DnnlProviderFactory() override {}
 
-  std::unique_ptr<IExecutionProvider> CreateProvider() override;
+  std::unique_ptr<Provider_IExecutionProvider> CreateProvider() override;
 
  private:
   bool create_arena_;
 };
 
-std::unique_ptr<IExecutionProvider> DnnlProviderFactory::CreateProvider() {
+std::unique_ptr<Provider_IExecutionProvider> DnnlProviderFactory::CreateProvider() {
   DNNLExecutionProviderInfo info;
   info.create_arena = create_arena_;
   return onnxruntime::make_unique<DNNLExecutionProvider>(info);
 }
 
-std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_Dnnl(int device_id) {
-  return std::make_shared<onnxruntime::DnnlProviderFactory>(device_id);
-  //TODO: This is apparently a bug. The consructor parameter is create-arena-flag, not the device-id
-}
+struct Dnnl_Provider : Provider {
+  std::shared_ptr<Provider_IExecutionProviderFactory> CreateExecutionProviderFactory(int device_id) override {
+    *reinterpret_cast<int*>(1234) = 5;
+    //TODO: This is apparently a bug. The consructor parameter is create-arena-flag, not the device-id
+    return std::make_shared<DnnlProviderFactory>(device_id);
```

Comment:
I actually cleaned this up in [https://github.com/microsoft/onnxruntime/pull/2850](https://github.com/microsoft/onnxruntime/pull/2850) but that hasn't been checked in yet. I was going to wait until 1.3 is done given it touches a lot of places.

Reply:
Awesome, should I remove the todo or leave it in?

Reply:
I'd leave it in but mention the PR that should address it once updated.

---
In reply to: [419816661](https://github.com/microsoft/onnxruntime/pull/3314#discussion_r419816661) [](ancestors = 419816661)

Reply:
Got it